File: src/test/java/game/ladder/domain/LevelTest.java
package game.ladder.domain;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class LevelTest {

    @Test
    public void getLevel_high() {
        final int high1 = Level.MAXIMUM_RATE;
        final int high2 = Level.MAXIMUM_RATE - 20;

        assertThat(Level.getLevel(high1)).isSameAs(Level.HIGH);
        assertThat(Level.getLevel(high2)).isSameAs(Level.HIGH);
    }

    @Test
    public void getLevel_middle() {
        final int middle1 = Level.MAXIMUM_RATE - 30;
        final int middle2 = Level.MAXIMUM_RATE - 21;

        assertThat(Level.getLevel(middle1)).isSameAs(Level.MIDDLE);
        assertThat(Level.getLevel(middle2)).isSameAs(Level.MIDDLE);
    }

    @Test
    public void getLevel_low() {
        final int low1 = Level.MAXIMUM_RATE - 50;
        final int low2 = Level.MINIMUM_RATE;

        assertThat(Level.getLevel(low1)).isSameAs(Level.LOW);
        assertThat(Level.getLevel(low2)).isSameAs(Level.LOW);
    }

    @Test(expected = IllegalArgumentException.class)
    public void getLevel_0_넘길때() {

        Level.getLevel(0);
    }

    @Test(expected = IllegalArgumentException.class)
    public void getLevel_101_넘길때() {

        Level.getLevel(101);
    }
}

File: src/test/kotlin/ladder/domain/LevelTest.kt
package ladder.domain

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.util.*

class LevelTest {
    @Test
    fun `문자열로 사다리 높이 가져오기`() {
        assertThat(Level.findLevel("상").getHeight()).isEqualTo(20)
    }

    @Test(expected = IllegalArgumentException::class)
    fun `없는 레벨 에러`() {
        Level.findLevel("초특등급")
    }

    @Test
    fun `레벨 가져오기`() {
        assertThat(Level.findLevel("상")).isEqualTo(Level.HIGH)
    }
}
